fs-base/src/main/java/com/palantir/giraffe/file/MoreFiles.java

```diff
@@ -569,6 +570,56 @@ public static boolean isEmpty(Path path) throws IOException {
     }
 
     /**
+     * A {@code FileVisitor} that builds a list of {@code Path} objects it visits.
+     *
+     * @author jyu
+     */
+    private static final class ListVisitor extends SimpleFileVisitor<Path> {
+        private final ImmutableList.Builder<Path> paths;
+
+        public ListVisitor() {
+            paths = new ImmutableList.Builder<>();
+        }
+
+        /**
+         * Adds the visited {@code Path} to the list of visited {@code Path} objects.
+         */
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
+                throws IOException {
+            paths.add(file);
+            return FileVisitResult.CONTINUE;
+        }
+
+        /**
+         * Returns the current list of visited {@code Path} objects.
+         *
+         * The returned path list can change between calls if this visitor visits any paths.
+         */
+        public List<Path> getPaths() {
+            return paths.build();
+        }
+    }
+
+    /**
+     * Given a {@code Path} representing a directory, retrieve the list of files
+     * and recursively contained in that {@code Path} as a {@code List}. The
+     * {@code Path} objects are obtained as if by resolving the name of the
+     * directory objects against the given {@code Path}.
```

Comment:
Ugh, I didn't realize the docs for `listDirectory` were this bad - I'll have to fix those. For this method, maybe something like:

"Gets a list of the files in the given directory and all sub-directories. The returned paths are relative to the given directory."